.devops/nix/package.nix

```diff
@@ -61,54 +66,13 @@ let
   pnameSuffix =
     strings.optionalString (suffices != [ ])
       "-${strings.concatMapStringsSep "-" strings.toLower suffices}";
-  descriptionSuffix =
-    strings.optionalString (suffices != [ ])
-      ", accelerated with ${strings.concatStringsSep ", " suffices}";
+  descriptionSuffix = strings.optionalString (
+    suffices != [ ]
+  ) ", accelerated with ${strings.concatStringsSep ", " suffices}";
 
   executableSuffix = effectiveStdenv.hostPlatform.extensions.executable;
 
-  # TODO: package the Python in this repository in a Nix-like way.
-  # It'd be nice to migrate to buildPythonPackage, as well as ensure this repo
-  # is PEP 517-compatible, and ensure the correct .dist-info is generated.
-  # https://peps.python.org/pep-0517/
-  #
-  # TODO: Package up each Python script or service appropriately, by making
-  # them into "entrypoints"
-  llama-python = python3.withPackages (
-    ps: [
-      ps.numpy
-      ps.sentencepiece
-    ]
-  );
-
-  # TODO(Green-Sky): find a better way to opt-into the heavy ml python runtime
-  llama-python-extra = python3.withPackages (
-    ps: [
-      ps.numpy
-      ps.sentencepiece
-      ps.tiktoken
-      ps.torchWithoutCuda
-      ps.transformers
-
-      # server bench
-      ps.matplotlib
-
-      # server tests
-      ps.openai
-      ps.behave
-      ps.prometheus-client
-
-      # for examples/pydantic-models-to-grammar-examples.py
-      ps.docstring-parser
-      ps.pydantic
-
-      # for scripts/compare-llama-bench.py
-      ps.gitpython
-      ps.tabulate
```

Comment:
I think these new optional dependencies added in #8341 should also be ported over in some way. (to `.devops/nix/python-scripts.nix`?)

(Just so you know, I use `nix develop .#default-extra` a lot in my workflow. What would be the equivalent after this gets merged?)

Reply:
That's a good question. What are your thoughts about:
1. Removing the passthru.shell/shell-extra construct
2. Adding a shell-only scope that yields a devShell with `inputsFrom` the main package, the llama-scripts package (plus dev dependencies)

cc @SomeoneSerge

Reply:
> 1. Removing the passthru.shell/shell-extra construct

I think this construct is not necessary, since `nix develop` can already build a shell from the dependencies of a package. So I agree with removing it.

> 2. Adding a shell-only scope that yields a devShell with `inputsFrom` the main package and the `llama-scripts` package (plus dev dependencies)

This sounds like the right thing to do. All I want is some name to give to `nix develop` to get all the Python dependencies + the C/C++ build dependencies of the default package, like `.#default-extra` currently seems to do.

Ideally, this would make devShells for each of the `llama.cpp` packages (`default-extra`, `vulkan-extra`, `cuda-extra`, etc.).

I think it should be doable to make theses `-extra` shells from `.devops/nix/devshells.nix` as you're suggesting (with `inputsFrom`) without using the `passthru` construct and without necessarily having to make the non-`extra` ones, because the python scripts are now in their own derivation and because `nix develop` seems to do the right thing even when `.devops/nix/devshells.nix` is commented-out, i.e. it makes a shell with the dependencies of the `.#vulkan` package when using `nix develop .#vulkan`.

Unfortunately, `nix develop` can only be given 1 name at a time.
Otherwise I would do `nix develop .#default .#python-scripts`, but this doesn't work, so `-extra` shells seem needed.

Reply:
@compilade apologies for the delayed turnaround for this. Please take a look at [0188318](https://github.com/ggerganov/llama.cpp/commit/01883187b519240c94c9fb156a870483256acd9a).

Reply:
@ditsuke Thanks for having implemented that. I've tried it, and it's nice that it also builds the `gguf` Python package and puts into the environment.

Something I've noticed, though, is that the `$PATH` now contains much more Python dependencies, even though they were also previously accessible to the Python interpreter before that. Not sure if that's a bad thing or not. Hopefully no Python package has a binary called `bash`, because given the search order, it would have priority.

It seems like some of the `gguf-py/scripts` are included in the `gguf` package, which means they are now in my `$PATH` when using `nix develop .#default-extra`. This can be nice to test if the Python build works properly, but it requires re-making the devShell at each change.

This is not a problem, because when directly testing the scripts, they insert the path of the local `gguf` library when `NO_LOCAL_GGUF` is not defined.

So my only remaining concern about this is the Python dependencies in the `$PATH`. But it's not a big problem. Actually, it might be useful because it makes stuff like `transformers-cli` and `huggingface-cli` available too.

I'll try to review this more deeply soon, and hopefully approving it.